### PR TITLE
Spike: Import QR Code via Image (EXPOSUREAPP-8358)

### PR DESCRIPTION
**Pull request by Kolpa**
### Description 

This draft PR contains the code written for the image qr code scanning spike, allowing the user to scan a qr code from a gallery image / pdf instead of having to use the device camera.

### This is currenlty very much in progress as the zxing library is currently not able to find any images in the provided bitmaps for unknown reasons

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/QRCodeFileParser.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import javax.inject.Inject
+
+@Reusable
+class QRCodeFileParser @Inject constructor(
+    private val dispatcherProvider: DispatcherProvider,
+    private val qrCodeBitmapProvider: QRCodeBitmapProvider,
+    private val qrCodeReader: QRCodeReader
+) {
+
+    suspend fun decodeQrCodeFile(fileUri: Uri): QRCodeParseResult = withContext(dispatcherProvider.IO) {
```
Comment (Kolpa) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/QRCodeFileParser.kt:24]: this is where the magic should happen 🧙

**Comment by harambasicluka** [comment]
Do not merge this, as it is a spike. We need a proper story to implement this.

**Review by mtwalli — APPROVED** [review]
LGTM

**Review by BMItr — APPROVED** [review]

**Comment by Ein-Tim** [comment]
Hi @mtwalli

Will this feature now be delivered with version 2.7 under Android? AFAIK iOS has not yet implemented this.

**Comment by mtwalli** [comment]
@Ein-Tim No, it just a preparation for the actual task. it is only accessible under `Test menu`.

**Comment by Ein-Tim** [comment]
@mtwalli Okay, thanks for the clarification!

**Pull request merged**